tests/src/main/java/com/hazelcast/stabilizer/tests/map/PredicatesTest.java

```diff
@@ -0,0 +1,347 @@
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.*;
+import com.hazelcast.query.Predicate;
+import com.hazelcast.query.SqlPredicate;
+import com.hazelcast.stabilizer.probes.probes.IntervalProbe;
+import com.hazelcast.stabilizer.test.TestContext;
+import com.hazelcast.stabilizer.test.TestRunner;
+import com.hazelcast.stabilizer.test.annotations.*;
+import com.hazelcast.stabilizer.tests.helpers.StringUtils;
+import com.hazelcast.stabilizer.test.utils.ThreadSpawner;
+import com.hazelcast.stabilizer.worker.Metronome;
+import com.hazelcast.stabilizer.worker.SimpleMetronome;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class PredicatesTest {
+
+    public static enum SerializationStrategy {
+        DATA_SERIALIZABLE,
+        JAVA_SERIALIZABLE,
+        IDENTIFIED_DATA_SERIALIZABLE,
+        PORTABLE,
+    }
+
+    private final static ILogger log = Logger.getLogger(PredicatesTest.class);
+    private final AtomicLong operations = new AtomicLong();
+    // properties
+    public int threadCount = 40;
+    public int keyLength = 10;
+    // number of keys per member
+    public int keyCount = 10000;
+    public int logFrequency = 10000;
+    public int performanceUpdateFrequency = 1;
+    public String name = "aaa";
+    public String sqlQuery = "name = " + name;
+    public String basename = "predicatesTest";
+    //default
+    public SerializationStrategy serializationStrategy = SerializationStrategy.PORTABLE;
+    public boolean customPredicate = false;
+    public IntervalProbe search;
+    public int intervalMs = 0;
+
+    private static final String[] names = {"aaa", "bbb", "ccc", "ddd", "eee", "fff", "ggg"};
+    private IMap<String, Employee> map;
+    private TestContext testContext;
+    private HazelcastInstance targetInstance;
+
+    @Setup
+    public void setup(TestContext testContext) throws Exception {
+        this.testContext = testContext;
+        this.targetInstance = testContext.getTargetInstance();
+        this.map = targetInstance.getMap(basename + "-" + testContext.getTestId());
+    }
+
+    @Teardown
+    public void teardown() throws Exception {
+        map.destroy();
+    }
+
+    //configured true to prevent map size changes from number of clients and members' effect.
+    @Warmup(global = true)
+    public void warmup() throws InterruptedException {
+        Random random = new Random();
+        for (int k = 0; k < keyCount; k++) {
+            int id = random.nextInt();
+            String key = StringUtils.generateString(keyLength);
+            String name = names[random.nextInt(names.length)];
+            Employee value = createEmployee(id,name);
+            map.put(key, value);
+        }
+        log.info("Map size is:" + map.size());
+    }
+
+    @Run
+    public void run() {
```

Comment:
Can you convert this test to the new workers approach we are using? So with the RunWithWorker annotation.